### DISCUSSION TITLE
Is it possible to have node-config resolve json refs within the local config file?

### DISCUSSION THREAD

**Discussion by damienbutt** [node-config/node-config#741] [Q&A] [answered]
Hi,

I'm using `node-config` with a project I'm working on and I have started using some JSON refs to reference back to some common properties in the file.

Is there a way to make `node-config` work with these? Currently as well as using `node-config`, I am using another package called `json-refs`, which has the ability to take the whole config object (using `config.util.toObject()`) and return a new object with any refs resolved.

 Ideally, I would like this to be handled by `node-config` so I can use the dot notation as normal using `config.get()`.

Any thoughts or ideas? Looking through the docs, there doesn't appear to be any plugin/hook system to have something run prior to retrieving the data. Unless I have overlooked something.

Thanks in advance,

D

**Comment by lorenwest** [comment]
That sounds like a valuable feature - anyone out there that wants to implement?

**Comment by markstos** [comment, marked as answer]
The `defer()` feature specifically for the JavaScript file format implements this. It works by allowing you declare config as a function that is then resolved after the final configuration is otherwise "rendered". In this way, configuration can reference values and properties that might can't defined later through the cascading resolution. 

The other way to handle this is with some third party solution solves the issue outside of node-config and renders out a JSON file, which `node-config` still reads.

If there are examples of solving this purely with JSON files, please share them.
